src/lib/components/TextInput/TextInput.tsx

```diff
@@ -79,11 +83,25 @@ export const TextInput = forwardRef<HTMLInputElement, TextInputProps>(
                 <Icon className={theme.field.icon.svg} />
               </div>
             )}
+
+            {renderIcon && !Icon && (
+              <div className={theme.field.icon.base} data-testid="rendered-icon">
+                {renderIcon(theme.field.icon.svg)}
+              </div>
+            )}
+
             {RightIcon && (
               <div data-testid="right-icon" className={theme.field.rightIcon.base}>
                 <RightIcon className={theme.field.rightIcon.svg} />
               </div>
             )}
+
+            {renderRightIcon && !RightIcon && (
+              <div data-testid="rendered-right-icon" className={theme.field.rightIcon.base}>
```

Comment:
A testid like `flowbite-textinput-righticon` would probably be more appropriate, but that's a nitpick, I don't really care that much if you change it.